# Covid-19 Data Analysis

### This Notebook aims to Analyze the publicly available COVID-19 data sourced by CDC, but publicly available on healthdata.gov. You can download the data from here https://healthdata.gov/dataset/provisional-covid-19-death-counts-sex-age-and-state



#### Importing the necessary python libraries

In [17]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
sns.set(style="ticks")
import plotly.express as px
import cufflinks as cf
cf.go_offline()

#### Load the data into Pandas Dataframe

In [4]:
df = pd.read_csv('./Provisional_COVID-19_Death.csv')

#### Get the shape of the Dataframe

In [6]:
df.shape

(1416, 13)

#### Get the information of the dataframe

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1416 entries, 0 to 1415
Data columns (total 13 columns):
Data as of                                  1416 non-null object
Start week                                  1416 non-null object
End Week                                    1416 non-null object
State                                       1416 non-null object
Sex                                         1416 non-null object
Age group                                   1416 non-null object
COVID-19 Deaths                             1141 non-null float64
Total Deaths                                1285 non-null float64
Pneumonia Deaths                            1089 non-null float64
Pneumonia and COVID-19 Deaths               1119 non-null float64
Influenza Deaths                            879 non-null float64
Pneumonia, Influenza, or COVID-19 Deaths    1063 non-null float64
Footnote                                    876 non-null object
dtypes: float64(6), object(7)
memory usage: 

#### Describe the Data on the dataframe

In [8]:
df.describe()

COVID-19 Deaths  Total Deaths  Pneumonia Deaths  \
count      1141.000000  1.285000e+03       1089.000000   
mean        803.774759  8.251070e+03        923.983471   
std        5813.512354  6.227365e+04       6469.131249   
min           0.000000  0.000000e+00          0.000000   
25%           0.000000  7.200000e+01          0.000000   
50%          20.000000  4.750000e+02         54.000000   
75%         173.000000  2.651000e+03        286.000000   
max      114741.000000  1.324958e+06     125868.000000   

       Pneumonia and COVID-19 Deaths  Influenza Deaths  \
count                    1119.000000        879.000000   
mean                      353.882931         57.146758   
std                      2535.556459        365.643951   
min                         0.000000          0.000000   
25%                         0.000000          0.000000   
50%                         1.000000          0.000000   
75%                        76.000000         21.000000   
max                     49623.000000       6492.000000   

       Pneumonia, Influenza, or COVID-19 Deaths  
count                               1063.000000  
mean                                1478.427093  
std                                10234.593491  
min                                    0.000000  
25%                                   12.000000  
50%                                   82.000000  
75%                                  419.500000  
max                               196538.000000

In [9]:
df.head(2)

Data as of  Start week    End Week          State  Sex     Age group  \
0  07/08/2020  02/01/2020  07/04/2020  United States  All  Under 1 year   
1  07/08/2020  02/01/2020  07/04/2020  United States  All     1-4 years   

   COVID-19 Deaths  Total Deaths  Pneumonia Deaths  \
0              9.0        7174.0              67.0   
1              7.0        1361.0              48.0   

   Pneumonia and COVID-19 Deaths  Influenza Deaths  \
0                            2.0              14.0   
1                            2.0              40.0   

   Pneumonia, Influenza, or COVID-19 Deaths Footnote  
0                                      88.0      NaN  
1                                      93.0      NaN

## Here is the brief overview about the data so far
### Dataset has about 1416 rows with 13 columns
### Time-series Data
#### Start of the week
#### End of the week
### User Demographics Include
#### - Age & Sex
### Mortality Types Include
#### - Influenza
#### - Pneumonia
#### - Influenza, Pneumonia & Covid-19
### Missing Data, Irrelevant Data & Timeseries Data
#### - Foot note column is irrelevant and will be dropped
#### - Right now Start week, End Week, and Data as of is in object, which needs to be converted into datetime format

In [12]:
df = df.drop(['Footnote'], axis=1)

## Viz-1
### This graph represents Covid Total Deaths by each state and broken down by Sex


In [56]:
fig = px.bar(df, x='State', y='Total Deaths', color='Sex', title='Total Deaths by each state and grouped by Sex')
fig.show()

## Viz-2
### This is a bar chart showing total COVID 19 deaths, broken down by State and Age group

In [60]:
fig = px.bar(df, x='State', y='Total Deaths', category_orders={'Age group': ['65-74 Years','75-84 Years']})
fig.show()